src/main/java/com/arcbees/bourseje/client/application/identification/IdentificationPresenter.java

```diff
@@ -35,11 +47,51 @@
     interface MyProxy extends ProxyPlace<IdentificationPresenter> {
     }
 
+    private final PlaceManager placeManager;
+    private final RestDispatch dispatcher;
+    private final VoteService voteService;
+
     @Inject
     IdentificationPresenter(
             EventBus eventBus,
             MyView view,
-            MyProxy proxy) {
+            MyProxy proxy,
+            PlaceManager placeManager,
+            RestDispatch dispatcher,
+            VoteService voteService) {
         super(eventBus, view, proxy, ApplicationPresenter.SLOT_MAIN);
+
+        this.placeManager = placeManager;
+        this.dispatcher = dispatcher;
+        this.voteService = voteService;
+
+        getView().setUiHandlers(this);
+    }
+
+    @Override
+    public void onSubmit(final String code) {
+        dispatcher.execute(voteService.useCode(code), new RestCallbackImpl<Void>() {
```

Comment:
Yes, but since there's like 3 calls to the server in the whole app, I didn't change what was already in place.